fix(apple): Clarify trace propagation targets (#9110)
getsentry/sentry-docs: docs/platforms/apple/common/usage/distributed-tracing/index.mdx

<PlatformContent includePath="distributed-tracing/how-to-use/" />

Remember that in order to propagate trace information through your whole distributed system, you have to use Sentry in all of the involved services and applications. Take a look at the respective SDK documentation to learn how distributed tracing can be enabled for each platform.

### tracePropagationTargets

Per default, the Cocoa SDK will attach the `sentry-trace` and `baggage` headers to all outgoing requests. You
can use the allowlist `tracePropagationTargets` to specify to which URLs the Cocoa SDK should send these headers.

> You can put string or regexes into the allowlist, and the SDK then only attaches the `sentry-trace` and `baggage`

Reviewer: ```suggestion
You can use strings or regexes in the allowlist. The SDK then only attaches the `sentry-trace` and `baggage`
```

Reviewer: that's much better, thank you @lizokm 🙏

headers to all outgoing requests whose destination contains a string in the list or matches a regex in the list.

> **The `tracePropagationTargets` option matches the entire request URL, not just the domain. Using stricter regex to match certain parts of the URL ensures that requests don't unnecessarily have additional headers attached.**

Reviewer: ```suggestion
**The `tracePropagationTargets` option matches the entire request URL, not just the domain. Use stricter regexes matching specific parts of the URL to ensure that requests don't have unnecessary or unwanted headers attached.**
```

Reviewer 2: I think it would be good to turn this into a note rather than bolding it.

> The default value of `tracePropagationTargets` is `['.*']`. This means that by default, tracing headers are attached to all requests.

Reviewer 2: ```suggestion
Since the value of `tracePropagationTargets` is `['.*']` by default, tracing headers are attached to all requests unless otherwise specified.
```
